skimage/exposure/exposure.py

```diff
@@ -120,10 +125,29 @@ def histogram(image, nbins=256, source_range='image', normalize=False):
     (array([107432, 154712]), array([ 0.25,  0.75]))
     """
     sh = image.shape
-    if len(sh) == 3 and sh[-1] < 4:
+    if len(sh) == 3 and sh[-1] < 4 and not multichannel:
         warn("This might be a color image. The histogram will be "
              "computed on the flattened image. You can instead "
-             "apply this function to each color channel.")
+             "apply this function to each color channel, or set "
+             "multichannel=True")
+
+    if multichannel:
+        channels = sh[-1]
+        hist = []
+        bin_centers = []
+        for chan in range(channels):
+            h, bc = _histogram(image[..., chan], nbins, source_range, normalize)
+            hist.append(h)
+        # Convert to numpy arrays
+        bin_centers = np.array(bc)
+        hist = np.array(hist)
```

Comment:
`bc` is already a numpy array, and hist should only be converted *after* the loop, not inside the loop.

Reply:
looks outside...

Reply:
Oops! The comment messed with my indentation detector! :sweat_smile: